packages/serverless/src/domain/use-case/database.ts

```diff
@@ -63,44 +70,39 @@ export async function getDatabases(
       }
     );
 
-    const collectionsCache: { [databaseName: string]: string[] } = {};
+    if (!settings?.length) {
+      return [];
+    }
+
+    const collectionsCache: Record<string, string[]> = {};
 
-    const databases: (Database | null)[] = await Promise.all(
+    const databases: Database[] = await Promise.all(
       settings.map(async (setting: DbSetting) => {
-        try {
-          const { databaseName, merkleHeight } = setting;
-
-          const dbInfo = databaseInfoMap[databaseName];
-          const databaseSize = dbInfo ? dbInfo.sizeOnDisk : null;
-
-          let collections = collectionsCache[databaseName];
-          if (!collections) {
-            collections =
-              await ModelDatabase.getInstance(databaseName).listCollections();
-            collectionsCache[databaseName] = collections;
-          }
-
-          return {
-            databaseName,
-            merkleHeight,
-            databaseSize,
-            collections,
-          } as Database;
-        } catch (error) {
-          logger.error(
-            `Error processing database ${setting.databaseName}:`,
-            error
-          );
-          return null;
-        }
+        const { databaseName, merkleHeight } = setting;
+        const dbInfo = databaseInfoMap[databaseName];
+        const databaseSize = dbInfo ? dbInfo.sizeOnDisk : null;
+
+        const collectionNames =
+          collectionsCache[databaseName] ||
+          (collectionsCache[databaseName] =
+            await ModelDatabase.getInstance(databaseName).listCollections());
+
+        const collections: Collection[] = await Promise.all(
```

Comment:
Use `@orochi-network/queue` and use Fill instead. Better error handling

Reply:
done